projects/golang/go/Makefile

```diff
@@ -118,6 +118,18 @@ tar-golang-sources:
 	mkdir -p $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)
 	cd /tmp/go-extracted/usr/lib/golang && rm -rf ./pkg/obj && tar czf $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz .
 
+.PHONY: generate-golang-checksum
+generate-golang-checksum: generate-rpm-checksum generate-archive-checksum
+
+.PHONY: generate-rpm-checksum
+generate-rpm-checksum:
+	mkdir -p $(VERSION_DIRECTORY)/checksums/$(GOOS)/$(ARCH_LOWER)
+	cd /tmp/go-extracted/usr/lib/golang && rm -rf ./pkg/obj && find . \( -not -name . \) -type f -exec cat {} + | sha256sum > $(VERSION_DIRECTORY)/checksums/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER)_RPM_CHECKSUM
+
+.PHONY: generate-archive-checksum
+generate-archive-checksum:
+	cd /tmp/go-extracted/usr/lib/golang && rm -rf ./pkg/obj && sha256sum $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz > $(VERSION_DIRECTORY)/checksums/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER)_ARCHIVE_CHECKSUM
+
 .PHONY: sync-artifacts-to-s3-dry-run
 sync-artifacts-to-s3-dry-run: check-env-release
```

Comment:
these two `make` targets, `sync-artifacts-to-s3-dry-run` and `sync-artifacts-to-s3`, are what actually pushes the locally build artifacts up to s3 (or tries to to detect errors, in the case of dry-run). If we output the sha files to the same directory alongside the corresponding artifacts, these will handle pushing them to s3 for us.